File: src/utils/formatting.py
def format_currency(value: float, symbol: str = "₪", decimal_places: int = 0) -> str:
    if abs(value) >= 1_000_000:
        return f"{symbol}{value / 1_000_000:,.1f}M"
    if abs(value) >= 1_000:
        return f"{symbol}{value / 1_000:,.{decimal_places}f}K"
    return f"{symbol}{value:,.{decimal_places}f}"


def format_currency_full(value: float, symbol: str = "₪") -> str:
    return f"{symbol}{value:,.0f}"


def format_percentage(value: float, decimal_places: int = 1) -> str:
    return f"{value:.{decimal_places}f}%"


def format_delta(value: float) -> str:
    sign = "+" if value > 0 else ""
    return f"{sign}{value:.1f}%"

File: frontend/src/lib/formatting.ts
export function formatCurrency(
  value: number,
  symbol = "\u20AA",
  decimalPlaces = 0
): string {
  const abs = Math.abs(value);
  if (abs >= 1_000_000) {
    return `${symbol}${(value / 1_000_000).toFixed(1)}M`;
  }
  if (abs >= 1_000) {
    return `${symbol}${(value / 1_000).toFixed(decimalPlaces)}K`;
  }
  return `${symbol}${value.toFixed(decimalPlaces)}`;
}

export function formatCurrencyFull(value: number, symbol = "\u20AA"): string {
  return `${symbol}${value.toLocaleString("en-US", { maximumFractionDigits: 0 })}`;
}

export function formatPercentage(value: number, decimalPlaces = 1): string {
  return `${value.toFixed(decimalPlaces)}%`;
}

export function formatDelta(value: number): string {
  const sign = value > 0 ? "+" : "";
  return `${sign}${value.toFixed(1)}%`;
}
